tests/framework/matcher/getter_test.go

```diff
@@ -0,0 +1,397 @@
+package matcher
+
+import (
+	"context"
+	"errors"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/testing"
+
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	cdifake "kubevirt.io/client-go/containerizeddataimporter/fake"
+	"kubevirt.io/client-go/kubecli"
+	kubevirtfake "kubevirt.io/client-go/kubevirt/fake"
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+
+	storagetypes "kubevirt.io/kubevirt/pkg/storage/types"
+
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+)
+
+var (
+	ctrl              *gomock.Controller
+	virtClient        *kubecli.MockKubevirtClient
+	originalGetClient func() kubecli.KubevirtClient
+)
+
+var _ = BeforeSuite(func() {
+	// Save the original getClient function to restore it later
+	originalGetClient = kubevirt.Client
+})
+
+var _ = AfterSuite(func() {
+	// Restore the original getClient function
+	getClient = originalGetClient
+	ctrl.Finish()
+})
```

Comment:
I am missing the reason why do we care about restoring the original vale, can you please elaborate?

Reply:
i thought about it and i added that in case this test suite will run with other suites, and I don't want it to affect other suites, but i could be wrong here

Reply:
turns out it was redundant.. so i removed it.